File: clients/dotnet/enums/TaxRuleTypeId.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace Avalara.AvaTax.RestClient
{
    /// <summary>
    /// 
    /// </summary>
    public enum TaxRuleTypeId
    {
        /// <summary>
        /// 
        /// </summary>
        RateRule,

        /// <summary>
        /// 
        /// </summary>
        RateOverrideRule,

        /// <summary>
        /// 
        /// </summary>
        BaseRule,

        /// <summary>
        /// 
        /// </summary>
        ExemptEntityRule,

        /// <summary>
        /// 
        /// </summary>
        ProductTaxabilityRule,

        /// <summary>
        /// 
        /// </summary>
        NexusRule,


    }
}

File: clients/php/enums/TaxRuleTypeId.php
<?php
/*
 * AvaTax Enum Class
 *
 * (c) 2004-2016 Avalara, Inc.
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */
namespace Avalara.AvaTax;

/**
 * @author Ted Spence <ted.spence@avalara.com>
 * @author Bob Maidens <bob.maidens@avalara.com>
 */class TaxRuleTypeId extends AvaTaxEnum 
{

    const RateRule = "RateRule";
    const RateOverrideRule = "RateOverrideRule";
    const BaseRule = "BaseRule";
    const ExemptEntityRule = "ExemptEntityRule";
    const ProductTaxabilityRule = "ProductTaxabilityRule";
    const NexusRule = "NexusRule";
}
